SPARCore/GADArticle.m

```diff
@@ -1,109 +1,86 @@
++ (NSArray <GADArticle *> *) articlesFromArray: (NSArray *)jsonArray {
     
-    NSMutableArray <GADArticle *> *articles = [[NSMutableArray alloc] init];
-    NSError *error = nil;
-    NSArray *jsonArray = [NSJSONSerialization JSONObjectWithData:json
-                                                         options:kNilOptions
-                                                           error:&error];
-    if (error != nil) {
-        NSLog(@"Error parsing JSON.");
-    }
+    NSMutableArray <GADArticle *> *articles = [NSMutableArray new];
     
     for (NSDictionary *element in jsonArray) {
-        //Map fields of element to fields of article
-        GADArticle *article = [[GADArticle alloc] init];
-        //datePublished field is a UNIX Timestamp number - converting to NSDate here
-        int timeStamp = (int)element[API_KEY_DATE_PUBLISHED];
-        article.articleId = element[API_KEY_ARTICLE_ID];
-        article.brief = element[API_KEY_BRIEF];
-        article.datePublished = [NSDate dateWithTimeIntervalSince1970: timeStamp];
-        article.headerImageURL = [NSURL URLWithString:element[API_KEY_HEADER_IMAGE]];
-        article.publicationId = element[API_KEY_PUBLICATION_ID];
-        article.title = element[API_KEY_TITLE];
-        if (element[API_KEY_CONTENT]) {
-            article.authors = element[API_KEY_AUTHORS];
-            article.content = element[API_KEY_CONTENT];
-        }
+        GADArticle *article = [self articleFromDictionary:element];
         [articles addObject:article];
     }
     return articles;
 }
 
-+ (NSURL *) baseQueryUrl {
-    NSURL *queryURL = [NSURL URLWithString:API_HOSTNAME];
-    queryURL = [NSURL URLWithString:API_PREFIX
-                      relativeToURL:queryURL];
-    return queryURL;
++ (GADArticle *) articleFromDictionary: (NSDictionary*)dict {
+    GADArticle *article = [GADArticle new];
+    //datePublished field is a UNIX Timestamp number - converting to NSDate here
+    int timeStamp = (int)dict[API_KEY_DATE_PUBLISHED];
+    article.datePublished = [NSDate dateWithTimeIntervalSince1970: timeStamp];
+    article.headerImageURL = [NSURL URLWithString:dict[API_KEY_HEADER_IMAGE]];
+    article.publication=[GADPublication new];
+    article.publication.publicationId = dict[API_KEY_PUBLICATION_ID];
+    article.articleId = dict[API_KEY_ARTICLE_ID];
+    article.title = dict[API_KEY_TITLE];
+    article.authors = dict[API_KEY_AUTHORS];
+    
+    return article;
 }
 
-+ (NSURL *) urlForArticlesWithPublicationId: (NSString *)publicationId{
-    NSURL *queryURL = [GADArticle baseQueryUrl];
-    queryURL = [NSURL URLWithString:API_PUBLICATION_PATH
-                      relativeToURL:queryURL];
-    queryURL = [NSURL URLWithString:publicationId
-                      relativeToURL:queryURL];
-    queryURL = [NSURL URLWithString:API_SUFFIX
-                      relativeToURL:queryURL];
-    return queryURL;
+- (void) fetchFullTextWithCompletion: (void(^_Nonnull)(GADArticle *_Nullable article,
```

Comment:
Is there no way to refactor this to use the `GADRemoteModel` method like everywhere else?